app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/RefreshController.kt

```diff
@@ -329,37 +328,57 @@ class RefreshController @Inject constructor(
     }
 
     private suspend fun Realm.fetchOnePage(scope: CoroutineScope, folder: Folder, direction: Direction): Set<Thread> {
-
-        val allUids = if (direction == Direction.TO_THE_FUTURE) folder.newMessagesUidsToFetch else folder.oldMessagesUidsToFetch
-        val (uidsToFetch, uidsRemaining) = if (allUids.count() > Utils.PAGE_SIZE) {
-            allUids.subList(0, Utils.PAGE_SIZE) to allUids.subList(Utils.PAGE_SIZE, allUids.count())
-        } else {
-            allUids to emptyList<Int>().toRealmList()
-        }
+        val (uidsToFetch, uidsRemaining) = computeUids(folder, direction)
 
         val impactedThreads = handleAddedUids(scope, folder, uidsToFetch)
```

Comment:
Is it not strange that on one line we talk about `uidsToFetch` and on the other we talk about `...AddedUids`?